common/libs/wallet/non-deterministic/helpers.ts

```diff
@@ -0,0 +1,90 @@
+import { IFullWallet } from 'ethereumjs-wallet';
+import { RawTransaction } from 'libs/transaction';
+import { signMessageWithPrivKey, signRawTxWithPrivKey } from 'libs/signing';
+import {
+  EncryptedPrivateKeyWallet,
+  MewV1Wallet,
+  PresaleWallet,
+  PrivKeyWallet,
+  UtcWallet
+} from './wallets';
+
+enum KeystoreTypes {
+  presale = 'presale',
+  utc = 'v2-v3-utc',
+  v1Unencrypted = 'v1-unencrypted',
+  v1Encrypted = 'v1-encrypted',
+  v2Unencrypted = 'v2-unencrypted'
+}
+
+interface ISignWrapper {
+  signRawTransaction(rawTx: RawTransaction): string;
+  signMessage(msg: string, addresss: string, date: string): string;
+  unlock();
+}
+
+type WrappedWallet = IFullWallet & ISignWrapper;
+
+export const signWrapper = (walletToWrap: IFullWallet): WrappedWallet =>
+  Object.assign(walletToWrap, {
+    signRawTransaction: (rawTx: RawTransaction) =>
+      signRawTxWithPrivKey(walletToWrap.getPrivateKey(), rawTx),
+    signMessage: (msg: string, address: string, date: string) =>
+      signMessageWithPrivKey(walletToWrap.getPrivateKey(), msg, address, date),
+    unlock: () => Promise.resolve()
+  });
+
+function determineKeystoreType(file: string): string {
+  const parsed = JSON.parse(file);
+  if (parsed.encseed) {
+    return KeystoreTypes.presale;
+  } else if (parsed.Crypto || parsed.crypto) {
+    return KeystoreTypes.utc;
+  } else if (parsed.hash && parsed.locked === true) {
+    return KeystoreTypes.v1Encrypted;
+  } else if (parsed.hash && parsed.locked === false) {
+    return KeystoreTypes.v1Unencrypted;
+  } else if (parsed.publisher === 'MyEtherWallet') {
+    return KeystoreTypes.v2Unencrypted;
+  } else {
+    throw new Error('Invalid keystore');
+  }
+}
+
+const isKeystorePassRequired = (file: string): boolean =>
+  determineKeystoreType(file) === KeystoreTypes.presale ||
+  KeystoreTypes.v1Encrypted ||
```

Comment:
Won't this always return true?

Reply:
It should return false on any unencrypted wallet

Reply:
Ah you're right, I should be checking it against the return value of `determineKeyStoreType` on each union